pkg/storage/mvcc_test.go

```diff
@@ -5248,6 +5248,58 @@ func TestResolveIntentWithLowerEpoch(t *testing.T) {
 	}
 }
 
+// TestTimeSeriesMVCCStats ensures that merge operations
+// result in expected increase in timeseries data
+func TestTimeSeriesMVCCStats(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	for _, engineImpl := range mvccEngineImpls {
+		t.Run(engineImpl.name, func(t *testing.T) {
+			engine := engineImpl.create()
+			defer engine.Close()
+			var ms = enginepb.MVCCStats{}
+
+			// Perform the same sequence of merges on the same key
+			// twice and record timeseries data
+			var firstSysBytes = ms.SysBytes
+			var firstLiveBytes = ms.LiveBytes
+			var firstValBytes = ms.ValBytes
+			if err := MVCCMerge(ctx, engine, &ms, testKey1, hlc.Timestamp{Logical: 1}, tsvalue1); err != nil {
+				t.Fatal(err)
+			}
+
+			var secondSysBytes = ms.SysBytes
+			var secondLiveBytes = ms.LiveBytes
+			var secondValBytes = ms.ValBytes
+			if err := MVCCMerge(ctx, engine, &ms, testKey1, hlc.Timestamp{Logical: 1}, tsvalue1); err != nil {
+				t.Fatal(err)
+			}
+
+			// Ensure timeseries metrics increases as expected
+			if isSysLocal(testKey1) {
```

Comment:
Since `testKey1` is not system-local, this may not be necessary. I think this would only make sense if we run this test against multiple test keys instead of just the one currently, which would be ideal to cover both conditions. What do you think?

Reply:
I'd be fine with using a specific testSysKey for this this that's constructed to be a system key. I'd also be fine with using testKey1 as is and simply testing {Live,Val}Bytes, we can skip this conditional given it's always going to be false.